exonum/src/blockchain/schema.rs

```diff
@@ -463,8 +464,16 @@ where
     }
 
     /// Changes the transaction status from `in_pool`, to `committed`.
-    pub(crate) fn commit_transaction(&mut self, hash: &Hash) {
-        self.transactions_pool().remove(hash);
+    pub(crate) fn commit_transaction(&mut self, hash: &Hash, tx: Signed<RawTransaction>) {
+        if !self.transactions().contains(hash) {
```

Comment:
Not a very clear case, what means if we are trying to commit a transaction that is not in `self.transactions()`?

Reply:
Because now we have transactions that stored only in cache during the round and added to storage after commit(or before the node is stopped).